vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/IntegerField.java

```diff
@@ -19,11 +19,52 @@
 import com.vaadin.flow.component.dependency.JsModule;
 import com.vaadin.flow.component.dependency.NpmPackage;
 import com.vaadin.flow.component.shared.HasThemeVariant;
+import com.vaadin.flow.data.binder.Binder;
 import com.vaadin.flow.function.SerializableFunction;
 
 /**
  * IntegerField is an extension of Text Field that only accepts integer numbers.
  *
+ * <h2>Validation</h2>
+ * <p>
+ * Integer Field comes with a built-in validation mechanism based on
+ * constraints. Validation is triggered whenever the user applies an input
+ * change, for example by pressing Enter or blurring the field. Programmatic
+ * value changes trigger validation as well. In eager and lazy value change
+ * modes, validation is also triggered on every character press with a delay
+ * according to the selected mode.
+ * <p>
+ * Validation verifies that the value is parsable into {@link Integer} and
+ * satisfies the specified constraints. If validation fails, the component is
+ * marked as invalid and an error message is displayed below the input.
+ * <p>
+ * The following constraints are supported:
+ * <ul>
+ * <li>{@link #setRequiredIndicatorVisible(boolean)}
+ * <li>{@link #setMin(double)}
+ * <li>{@link #setMax(double)}
+ * <li>{@link #setStep(double)}
```

Comment:
Please note that `IntegerField` has overrides to use `int` instead of `double`:

```suggestion
 * <li>{@link #setMin(int)}
 * <li>{@link #setMax(int)}
 * <li>{@link #setStep(int)}
```

Reply:
Oh, indeed, thanks!